Samples/SimpleIslandApp/SimpleIslandApp.cpp

```diff
@@ -0,0 +1,224 @@
+﻿// SimpleIslandApp.cpp : Defines the entry point for the application.
+//
+
+#include "pch.h"
+#include "SimpleIslandApp.h"
+
+#include <App.xaml.h>
+#include <MainPage.h>
+
+namespace winrt
+{
+    using namespace winrt::Microsoft::UI;
+    using namespace winrt::Microsoft::UI::Dispatching;
+    using namespace winrt::Microsoft::UI::Xaml;
+    using namespace winrt::Microsoft::UI::Xaml::Hosting;
+    using namespace winrt::Microsoft::UI::Xaml::Markup;
+}
+
+#define MAX_LOADSTRING 100
+
+// Global Variables:
+HINSTANCE hInst;                                // current instance
+WCHAR szTitle[MAX_LOADSTRING];                  // The title bar text
+WCHAR szWindowClass[MAX_LOADSTRING];            // the main window class name
+
+HWND g_button1{ NULL };
+HWND g_button2{ NULL };
+winrt::DesktopWindowXamlSource g_desktopWindowXamlSource {nullptr};
+
+// Forward declarations of functions included in this code module:
+ATOM                MyRegisterClass(HINSTANCE hInstance);
+BOOL                InitInstance(HINSTANCE, int);
+LRESULT CALLBACK    WndProc(HWND, UINT, WPARAM, LPARAM);
+INT_PTR CALLBACK    About(HWND, UINT, WPARAM, LPARAM);
+
+int APIENTRY wWinMain(_In_ HINSTANCE hInstance,
+                     _In_opt_ HINSTANCE hPrevInstance,
+                     _In_ LPWSTR    lpCmdLine,
+                     _In_ int       nCmdShow)
+{
+    UNREFERENCED_PARAMETER(hPrevInstance);
+    UNREFERENCED_PARAMETER(lpCmdLine);
+
+    winrt::init_apartment(winrt::apartment_type::single_threaded);
+
+    // We must start a DispatcherQueueController before we can create an island or use Xaml.
+    auto dispatcherQueueController {winrt::DispatcherQueueController::CreateOnCurrentThread()};
+
+    // Create our custom Xaml App object.
+    auto simpleIslandApp {winrt::make<winrt::SimpleIslandApp::implementation::App>()};
+
+    // Initialize global strings
+    LoadStringW(hInstance, IDS_APP_TITLE, szTitle, MAX_LOADSTRING);
```

Comment:
I'd make szTitle and szWindowClass statics and move their definitions and loads into InitInstance for better encapsulation

Reply:
Sure, I'll move them to stack variables.

Reply:
oh right - they don't even need to be local statics